pkg/updater/version_test.go

```diff
@@ -199,71 +169,69 @@ func TestCheckAppVersion(t *testing.T) {
 
 func TestCheckAppVersionAsync(t *testing.T) {
 	t.Parallel()
-	sampleAppResponse, err := json.Marshal(AppResponse{
+
+	testAppResponse := &AppResponse{
 		AppID:          "sample_app_1",
 		AppName:        "Sample App 1",
 		AppRepoURL:     "https://github.com/abcxyz/sample_app_1",
 		CurrentVersion: "1.0.0",
-	})
-	if err != nil {
-		t.Errorf("failed to encode json %v", err)
 	}
 
-	// Add 10ms of latency to response to allow testing timeout.
-	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-		time.Sleep(10 * time.Millisecond)
-
-		if !strings.HasSuffix(r.URL.Path, "sample_app_1/data.json") {
-			w.WriteHeader(http.StatusNotFound)
-			fmt.Fprintln(w, http.StatusText(http.StatusNotFound))
-			return
-		}
-
-		w.WriteHeader(http.StatusOK)
-		fmt.Fprintf(w, "%s\n", string(sampleAppResponse))
-	}))
-	t.Cleanup(ts.Close)
-
 	cases := []struct {
-		name          string
-		timeoutMillis int
-		want          string
-		wantErr       string
+		name    string
+		timeout time.Duration
+		want    string
+		wantErr string
 	}{
 		{
 			name: "happy_path_no_error",
 			want: `Sample App 1 version 1.0.0 is available at [https://github.com/abcxyz/sample_app_1]. Use SAMPLE_APP_1_IGNORE_VERSIONS="1.0.0" (or "all") to ignore.`,
 		},
 		{
-			name:          "happy_path_no_error_provided_timeout",
-			timeoutMillis: 4000,
-			want:          `Sample App 1 version 1.0.0 is available at [https://github.com/abcxyz/sample_app_1]. Use SAMPLE_APP_1_IGNORE_VERSIONS="1.0.0" (or "all") to ignore.`,
+			name:    "happy_path_no_error_provided_timeout",
+			timeout: 4 * time.Second,
+			want:    `Sample App 1 version 1.0.0 is available at [https://github.com/abcxyz/sample_app_1]. Use SAMPLE_APP_1_IGNORE_VERSIONS="1.0.0" (or "all") to ignore.`,
 		},
 		{
-			name:          "time_out_causes_error",
-			timeoutMillis: 9, // test server adds 10ms of latency, so this will timeout.
-			wantErr:       "context deadline exceeded",
+			name:    "time_out_causes_error",
+			timeout: 1 * time.Nanosecond,
+			wantErr: "context deadline exceeded",
 		},
 	}
-	for _, tc := range cases {
-		tc := tc
 
+	for _, tc := range cases {
 		t.Run(tc.name, func(t *testing.T) {
 			t.Parallel()
 
+			ts := httptest.NewServer(func() http.Handler {
+				mux := http.NewServeMux()
+				mux.HandleFunc("GET /sample_app_1/data.json", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Why have the mux in this case, if there is only one route? Is it so we catch wrong method / path by returning a 404?

Reply:
Right - there was a test for the 404 path, so this ensures we are hitting the right path (and HTTP verb).